app/src/main/java/ca/llamabagel/transpo/data/SearchRepository.kt

```diff
@@ -4,29 +4,80 @@
 
 package ca.llamabagel.transpo.data
 
+import ca.llamabagel.transpo.BuildConfig.MAPBOX_KEY
+import ca.llamabagel.transpo.R
+import ca.llamabagel.transpo.data.OttawaBoundaries.MAX_LAT
+import ca.llamabagel.transpo.data.OttawaBoundaries.MAX_LNG
+import ca.llamabagel.transpo.data.OttawaBoundaries.MIN_LAT
+import ca.llamabagel.transpo.data.OttawaBoundaries.MIN_LNG
 import ca.llamabagel.transpo.data.db.TransitDatabase
+import ca.llamabagel.transpo.di.StringsGen
 import ca.llamabagel.transpo.ui.search.viewholders.SearchResult
+import com.mapbox.api.geocoding.v5.MapboxGeocoding
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.withContext
 import javax.inject.Inject
 
-class SearchRepository @Inject constructor(private val database: TransitDatabase) {
-    suspend fun getStops(query: String): List<SearchResult> = withContext(Dispatchers.IO) {
-        val stops = database.stopQueries.getStopsByName("$query*").executeAsList()
-        val routes = database.routeQueries.getRoutes("$query%").executeAsList()
+object OttawaBoundaries {
+    const val MIN_LAT = 45.203039
+    const val MIN_LNG = -76.227729
+    const val MAX_LAT = 45.513359
+    const val MAX_LNG = -75.351405
+}
+
+class SearchRepository @Inject constructor(private val database: TransitDatabase, private val strings: StringsGen) {
+
+    suspend fun getSearchResults(query: String): List<SearchResult> = withContext(Dispatchers.IO) {
+        if (query.isEmpty()) return@withContext emptyList<SearchResult>()
+
+        val stops = getStops(query)
```

Comment:
```suggestion
        getStops(query).takeIf { it.isNotEmpty() }?.let { /* Add header and append to search results */ }
```

This could be made more idiomatic by using a `takeIf` followed by a `let` block where the  header and results can be added to the output list.